tests/src/test/java/io/streamnative/pulsar/handlers/kop/KopEventManagerTest.java

```diff
@@ -0,0 +1,214 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.streamnative.pulsar.handlers.kop;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import com.google.common.collect.Lists;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.ConsumerGroupDescription;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.ConsumerGroupState;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+
+public class KopEventManagerTest extends KopProtocolHandlerTestBase {
+    private AdminClient adminClient;
+    private String broker;
+    private final String topic1 = "test-topic1";
+    private final String topic2 = "test-topic2";
+    private final String topic3 = "test-topic3";
+
+    @BeforeMethod
+    @Override
+    protected void setup() throws Exception {
+        super.internalSetup();
+        final EndPoint plainEndPoint = getPlainEndPoint();
+        this.broker = plainEndPoint.getHostname() + ":" + plainEndPoint.getPort();
+        Properties adminPro = new Properties();
+        adminPro.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, broker);
+        this.adminClient = AdminClient.create(adminPro);
+    }
+
+    @AfterMethod
+    @Override
+    protected void cleanup() throws Exception {
+        adminClient.close();
+        super.internalCleanup();
+    }
+
+    @Test
+    public void testGroupState() throws Exception {
```

Comment:
It took about 1 minute to run this test in my local env. Is it necessary? Please and the timeout to `@Test` and reduce the test time. It's better shorter than 10 seconds even 5 seconds. If the test is unstable for shorter time, there would be some serious problems with the implementation.